crates/trie/sparse/src/lib.rs

```diff
@@ -1,6 +1,24 @@
 //! The implementation of sparse MPT.
 
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc as std;
+
+/// A macro that expands to a trace! call when std feature is enabled,
+/// and does nothing when `no_std` is enabled.
+#[macro_export]
+macro_rules! trace {
+    ($($args:tt)*) => {
+        #[cfg(feature = "std")]
+        reth_tracing::tracing::trace!($($args)*);
+        #[cfg(not(feature = "std"))]
+        {
+            // Currently in no_std, trace! is a noop
+        };
+    };
+}
```

Comment:
We could put this in `rust-tracing` and disable tracing etc in no-std

Reply:
tracing supports `no_std` (see https://github.com/tokio-rs/tracing/blob/dfc2c8b81889f1bc65f053e574de32ec79b72ce1/tracing/Cargo.toml#L64), why is this needed?